internal/manifest/xz.go

```diff
@@ -0,0 +1,53 @@
+package manifest
+
+import (
+	"github.com/osbuild/osbuild-composer/internal/artifact"
+	"github.com/osbuild/osbuild-composer/internal/osbuild"
+)
+
+// The XZ pipeline compresses a raw image file using xz.
+type XZ struct {
+	Base
+	Filename string
+
+	imgPipeline *RawOSTreeImage
```

Comment:
I wanted to do this but it would need a new kind of interface I think, an interface for pipelines that generate a single file.  I'd rather do it in a followup (or when we end up needing it).

Reply:
I may be wrong, but I don't think that you would (at least based on what I looked up during the review).

The `manifest.Pipeline` interface has `Export()` method, which returns `artifact.Artifact`, which is a struct implementing  a `Filename()` method. That looks like all you need in this stage 🤔

Reply:
Also we will need the XZ pipeline for `ec2` as well as `azure-rhui` images if we ever end up porting RHEL definitions to this approach.